File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    mode = 0
    ret = ""
    for idx, v in enumerate(code):
        if mode == 0:
            if code[idx] != "1":
                if idx%2 == 0:
                    ret += v
            else:        
                mode = 1
        elif mode == 1:
            if code[idx] != "1":
                if idx%2 != 0:
                    ret += v
            else:
                mode = 0
                    

    return ret if ret != "" else "EMPTY"

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        StringBuilder sb = new StringBuilder();
        int mode = 0;
        for (int i = 0; i < code.length(); i++) {
            char current = code.charAt(i);
            if (current == '1') {
                mode = mode == 0 ? 1 : 0;
                continue;
            }

            if (i % 2 == mode) {
                sb.append(current);
            }
        }
        return sb.length() == 0 ? "EMPTY" : sb.toString();
    }
}
